### PR TITLE
Walter docker update

### PR DESCRIPTION
**Pull request by warniaha**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/daos/docker/build-dev.sh [hunk 1]
```diff
@@ -51,35 +59,47 @@ function get_s3_dockerfile()
...
+{
+    set +e
+    daosrocky=`docker images | grep -o "daos-rocky"`
+    set -e
+    if [[ ! $daosrocky == 'daos-rocky' ]]; then
+        docker build "https://github.com/daos-stack/daos.git#${DAOS_BRANCH}" -f utils/docker/Dockerfile.el.8 -t daos-rocky
+        if [[ ! $? == 0 ]]; then exit 1; fi
+    fi
+    pushd daos
+    docker build . -t "daos-single-$image_type"
```
Comment (zalsader) [inline src/daos/docker/build-dev.sh:85]: Where is `$image_type` defined?

File: src/daos/docker/ceph/Dockerfile [hunk 2]
```diff
@@ -11,6 +11,28 @@ LABEL maintainer="seagate-daos@seagate.com"
...
+
+RUN sudo dnf install dnf-plugins-core -y
+RUN sudo dnf config-manager --set-enabled powertools
+RUN sudo dnf -y module enable javapackages-tools
+RUN sudo dnf install epel-release -y
+RUN sudo dnf install gcc-toolset-11 -y
+# xz is needed to decompress the tar file
+RUN sudo dnf install xz -y
+RUN sudo dnf install wget -y
+RUN bash -c "echo "source scl_source enable gcc-toolset-11" >> ~/.bashrc"
```
Comment (zalsader) [inline src/daos/docker/ceph/Dockerfile]: Does this get run too?

File: src/daos/docker/ceph/redhat.repo [hunk 3]
```diff
@@ -0,0 +1,96 @@
+[EOS_Puppet6_puppet6-el8]
```
Comment (zalsader) [inline src/daos/docker/ceph/redhat.repo:1]: Does this need to be in the repo? Is it not on the ssc machines?

**Review by zalsader — APPROVED** [review]

**Comment by zalsader** [comment]
Did you get a chance to test it or will it be tested automatically?

File: src/daos/docker/build-dev.sh (hunk 1 above)
Comment (warniaha) [reply src/daos/docker/build-dev.sh:85]: just a few lines down
`image_type="host"
if [[ ! $CEPH_BRANCH == $default_branch ]]; then
    image_type="dev"
fi
`

File: src/daos/docker/ceph/redhat.repo (hunk 3 above)
Comment (warniaha) [reply src/daos/docker/ceph/redhat.repo:1]: I basically took the repo file from one of the machines that had it and changed the https requests to http (with Tim's help).
I didn't look at the contents to pick and choose what to include, I just included everything from that Seagate repo.

File: src/daos/docker/ceph/Dockerfile (hunk 2 above)
Comment (warniaha) [reply src/daos/docker/ceph/Dockerfile]: I had tested it on the SSC 1901 machine and it didn't fail.  When running it on the SSC 3251 machine it did fail.  I was just looking at it.  Good catch.  I also used the quotes incorrectly, I'm not sure what actually got appended

**Review by tshaffe1 — APPROVED** [review]

**Comment by mssawant** [comment]
Hi @warniaha, @zalsader, do we have a commit message template that we follow? It would be great to describe the problem that we are trying to solve and fix.

**Pull request merged**